File: repeatedString.js
function repeatedString(s, n) {
    let x = s.split('a').length - 1;
    console.log(x);

    // let string = s;
    // let counter = 0;
    // for (let i = 0; i < n; i++) {
    //     if (string.length < n) {
    //         string = string + s;
    //     }
    //     if (string[i] == "a") {
    //         counter = counter + 1;
    //     }

    // }


    // return counter;

}

console.log(repeatedString("abaababa", 10));

File: ruby/repeated_string.rb
# frozen_string_literal: true

def repeatedString(s, n)
  if !s.include? 'a'
    0
  elsif only_a?(s)
    n
  else
    counter = 0
    repeats = s.length % n
    substring = n - s.length * repeats
    substring.times { |x| counter += 1 if s[x] == 'a' }
    counter += s.count('a') * repeats
  end
end

def only_a?(s)
  cnt = 0
  chars = s.split('')
  chars.each do |x|
    cnt += 1 if x == 'a'
  end
  return true if cnt == s.length

  false
end

puts only_a?('aaaabaaa')

puts repeatedString('aba', 10)
